packages/flame/test/components/position_component_test.dart

```diff
@@ -105,6 +105,36 @@ void main() {
       expect(component.containsPoint(point), true);
     });
 
+    test('component with anchor topLeft contains point on edge', () {
+      final size = Vector2(2.0, 2.0);
+      final Hitbox component = MyHitboxComponent();
+      component.position = Vector2(-1, -1);
+      component.anchor = Anchor.topLeft;
+      component.size = size;
+      final hitbox = HitboxRectangle();
+      component.addShape(hitbox);
+
+      expect(component.containsPoint(Vector2(1, 1)), true);
+      expect(component.containsPoint(Vector2(1, -1)), true);
+      expect(component.containsPoint(Vector2(-1, -1)), true);
+      expect(component.containsPoint(Vector2(-1, 1)), true);
+    });
+
+    test('component with anchor bottomRight contains point on edge', () {
+      final size = Vector2(2.0, 2.0);
+      final Hitbox component = MyHitboxComponent();
+      component.position = Vector2(1, 1);
+      component.anchor = Anchor.bottomRight;
+      component.size = size;
+      final hitbox = HitboxRectangle();
+      component.addShape(hitbox);
+
+      expect(component.containsPoint(Vector2(1, 1)), true);
+      expect(component.containsPoint(Vector2(1, -1)), true);
+      expect(component.containsPoint(Vector2(-1, -1)), true);
+      expect(component.containsPoint(Vector2(-1, 1)), true);
```

Comment:
maybe some expects outside that yield false would be good as well?

Reply:
specifically in a point that would be if the anchor was not applied

Reply:
Added a test for that